### PR TITLE
Refactor e2e kubernetes reporter

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
- Move to separate package
- Extract common code
- Ensure the reporting fires right after an It() block instead of (current behavior) after other AfterEach blocks.

This also fixes broken reporting when failures occurred on both the node 1 parallel/serial run

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/tests_suite_test.go [hunk 1]
```diff
@@ -166,4 +175,13 @@ func BuildTestSuite() {
 			Expect(err).ToNot(HaveOccurred())
 		}
 	})
+
+	JustAfterEach(func() {
```
Comment (akalenyu) [inline tests/tests_suite_test.go:179]: This will make sure we get artifacts that got deleted in AfterEach blocks
/cc @alromeros

**Review by awels — APPROVED** [review]
Just a note, its a battle between should we . import gingko and gomega or not. I see both, and the static linters will complain if we do. I am super passionate about either one, but we should be consistent, and we are not.

**Comment by akalenyu** [comment]
> Just a note, its a battle between should we . import gingko and gomega or not. I see both, and the static linters will complain if we do. I am super passionate about either one, but we should be consistent, and we are not.

Definitely don't want to make it worse. updated to dot import

**Review by awels — APPROVED** [review]

**Comment by alromeros** [comment]
> Ensure the reporting fires right after an It() block instead of (current behavior) after other AfterEach blocks
Thanks for the PR, Alex. This will be really useful in some cases.
/lgtm

**Pull request merged**